lib/jobs/ipmi-sdr-alert-job.js

```diff
@@ -48,55 +48,103 @@ function ipmiSdrPollerAlertJobFactory(
 
     IpmiSdrPollerAlertJob.prototype._determineAlert = function _determineAlert(data) {
         return waterline.workitems.needByIdentifier(data.workItemId)
-        .then(function(workitem) {
-
+        .then(function (workitem) {
             var conf = workitem.config;
-            if(_.has(conf, 'inCondition')) {
-                conf.inCondition = _.transform(conf.inCondition, function(result, val, key) {
+            if (_.has(conf, 'inCondition')) {
+                conf.inCondition.discrete = _.transform(conf.inCondition.discrete,
+                                                        function (result, val, key) {
+                    result[key.replace(/_/ig, '.')] = val;
+                });
+                conf.inCondition.threshold = _.transform(conf.inCondition.threshold,
+                                                         function (result, val, key) {
                     result[key.replace(/_/ig, '.')] = val;
                 });
             } else {
-                conf.inCondition = {};
+                var inCondition = {
+                    'discrete': {},
+                    'threshold': {}
+                };
+                conf.inCondition = inCondition;
             }
             // Set the pollerName var
             data.pollerName = "sdr";
-
-            var alerts = _.transform(data.sdr, function(results, val) {
+            var alerts = _.transform(data.sdr, function (results, val) {
                 var alertObj;
-                var sensorKey = val['Sensor Id'].replace(/_/ig, '.');
-
-                /* publish an alert if an active fault is detected (inCondition asserted) or
-                 * if a fault has just transitioned from active to inactive (inCondition is not
-                 * asserted, but conf.inCondition is).
-                 */
-                var unavailableStatuses = ['Not Available', 'ns', 'No Reading',
-                    'na','disabled', 'Disabled', 'Not Readable'];
-
-                var inCondition = val.Status !== 'ok' &&
-                    !_.contains(unavailableStatuses, val.Status);
-
-                var doAlert = inCondition || conf.inCondition[sensorKey];
-
-                if (doAlert) {
-                    alertObj = _.omit(data, 'sdr');
-                    alertObj.reading = val;
-                    alertObj.inCondition = inCondition;
-                    alertObj.node = data.node;
-                    conf.inCondition[sensorKey] = inCondition;
-                    results.push(alertObj);
+                var sensorKey = val.sensorId.replace(/_/ig, '.');
+                var sdrType = val.sdrType.replace(/_/ig, '.');
+
+                //Check if sdrType is Discrete/Threshold
+                if (sdrType === 'Discrete') {
+                    var statesAsserted = _.has(val, 'statesAsserted') ? val.statesAsserted : [];
```

Comment:
Just a suggestion: it would be a little more readable to separate each of these if/else bodies into functions.